src/server/pfs/server/master.go

```diff
@@ -228,61 +227,99 @@ func (d *driver) manageRepo(ctx context.Context, ring *consistenthashing.Ring, r
 	})
 }
 
-func (d *driver) manageBranches(ctx context.Context, repoKey string) error {
+type cronTrigger struct {
+	cancel context.CancelFunc
+	spec   string
+}
+
+func (d *driver) manageBranches(ctx context.Context, repoPair pfsdb.RepoPair) error {
+	repoKey := repoPair.RepoInfo.Repo.Key()
 	ctx, cancel := pctx.WithCancel(ctx)
 	defer cancel()
-	type cronTrigger struct {
-		cancel context.CancelFunc
-		spec   string
-	}
-	cronTriggers := make(map[string]*cronTrigger)
+	cronTriggers := make(map[pfsdb.BranchID]*cronTrigger)
 	defer func() {
 		for _, ct := range cronTriggers {
 			ct.cancel()
 		}
 	}()
-	return d.branches.ReadOnly(ctx).WatchByIndexF(pfsdb.BranchesRepoIndex, repoKey, func(ev *watch.Event) error {
-		if ev.Type == watch.EventError {
-			return ev.Err
+	watcher, err := postgres.NewWatcher(d.env.DB, d.env.Listener, path.Join(randutil.UniqueString(d.prefix), "manageBranches", repoKey),
+		fmt.Sprintf("%s%d", pfsdb.BranchesRepoChannelName, repoPair.ID))
```

Comment:
We should create a function in pfsdb for the channel name, or a constructor for the watcher in pfsdb.